runtime/src/installed_scheduler_pool.rs

```diff
@@ -70,6 +85,29 @@ impl BankWithScheduler {
         self.inner.bank.clone()
     }
 
+    pub fn has_installed_scheduler(&self) -> bool {
+        self.inner.scheduler.read().unwrap().is_some()
+    }
+
+    // 'a is needed; anonymous_lifetime_in_impl_trait isn't stabilized yet...
+    pub fn schedule_transaction_executions<'a>(
+        &self,
+        transactions: &[SanitizedTransaction],
+        transaction_indexes: impl Iterator<Item = &'a usize>,
+    ) {
```

Comment:
sorry I missed this on my previous pass. Could you explain/justify this interface? There's the comment on `'a` which drew my attention.

It feels a bit odd to have `transactions` be a slice, and `transaction_indexes` be an iterator.
AFAICT, the source of both of these is `TransactionBatchWithIndexes` which should be able to get us a slice for both these fields. Which makes me wonder why we need/want transaction_indexes to be an iterator.

I can see iterators are more generic than the slice; if that's the case, why not something like:

```rust
pub fn schedule_transaction_executions(
  &self,
  transactions_with_indexes: impl IntoIterator<Item = (...)>
) {
...
}
```

Reply:
you're right. i just simplified and i like this: 5d856f99af9f0670b5334ac3601f83d882323753